### ISSUE TITLE
MailRu synchronization fails with "panic: Invalid UInt64 -1" in addFileMetaData()

### ISSUE THREAD

**Issue by dmak** [rclone/rclone#6547]
I am using rclone v1.60.0 in Linux x64. When running synchronization with mailru cloud the following error occurs:
```
$ rclone sync -vvv --dump headers,bodies --track-renames ~/cloud mailru:

2022/11/07 15:09:47 DEBUG : HTTP REQUEST (req 0xc000247f00)
2022/11/07 15:09:47 DEBUG : PUT /upload/?client_id=cloud-win&token=a201d82985197c86a6ac4f636145b8d0ad76333737363830 HTTP/1.1
Host: cld-uploader8.cloud.mail.ru
User-Agent: rclone/v1.60.0
Content-Length: 1970770
Accept: */*
Accept-Encoding: gzip
...
2022/11/07 15:09:47 DEBUG : >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
2022/11/07 15:09:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
2022/11/07 15:09:48 DEBUG : HTTP RESPONSE (req 0xc000247f00)
2022/11/07 15:09:48 DEBUG : HTTP/1.1 201 Created
Content-Length: 40
Connection: keep-alive
Content-Type: text/plain
Date: Mon, 07 Nov 2022 14:09:48 GMT
Server: nginx/1.20.1
X-Host: cld-uploader8
X-Req-Id: rUkT5F7kzD
X-Req-Id: rUkT5F7kzD
X-Server: Streamer/1.0
X-Started: 1667830187.87987

F8F7DEC4EE581EF8E03BCD672BDD840B9AF72F29
2022/11/07 15:09:48 DEBUG : <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
panic: Invalid UInt64 -1

goroutine 73 [running]:
github.com/rclone/rclone/backend/mailru/api.(*BinWriter).WritePu64(0xc0006b97e8?, 0xc000d373b0?)
        github.com/rclone/rclone/backend/mailru/api/helpers.go:67 +0xe8
github.com/rclone/rclone/backend/mailru.(*Object).addFileMetaData(0xc0013c6420, {0x2334130?, 0xc000cf1280}, 0x1)
        github.com/rclone/rclone/backend/mailru/mailru.go:2061 +0x33c
github.com/rclone/rclone/backend/mailru.(*Object).Update(0xc0013c6420, {0x2334130, 0xc000cf1280}, {0x231f2a0, 0xc00238c600}, {0x7f7538811318, 0xc0011d3ec0}, {0xc002166b90, 0x1, 0x1})
        github.com/rclone/rclone/backend/mailru/mailru.go:1737 +0xf26
github.com/rclone/rclone/backend/mailru.(*Fs).Put(0xc0005c31e0, {0x2334130, 0xc000cf1280}, {0x231f2a0, 0xc00238c600}, {0x7f7538811318?, 0xc0011d3ec0?}, {0xc002166b90, 0x1, 0x1})
        github.com/rclone/rclone/backend/mailru/mailru.go:1598 +0x1d6
github.com/rclone/rclone/fs/operations.Copy({0x2334130, 0xc000cf1280}, {0x2345f40, 0xc0005c31e0}, {0x0?, 0x0?}, {0xc000ea0270, 0x2c}, {0x2345ed0, 0xc0011d3ec0})
        github.com/rclone/rclone/fs/operations/operations.go:501 +0x1b6e
github.com/rclone/rclone/fs/sync.(*syncCopyMove).pairCopyOrMove(0xc0008c98c0, {0x2334130, 0xc000cf1280}, 0x0?, {0x2345f40, 0xc0005c31e0}, 0x0?, 0x0?)
        github.com/rclone/rclone/fs/sync/sync.go:423 +0x1bb
created by github.com/rclone/rclone/fs/sync.(*syncCopyMove).startTransfers
        github.com/rclone/rclone/fs/sync/sync.go:450 +0x45
```

**Comment by ncw** [comment]
I had an experiment and it seems mailru does support files before 1970-01-01

So please could you try this and see if it helps

[v1.61.0-beta.6534.de73fdb57.fix-6547-mailru-times](https://beta.rclone.org/branch/fix-6547-mailru-times/v1.61.0-beta.6534.de73fdb57.fix-6547-mailru-times/) on branch [fix-6547-mailru-times](https://github.com/rclone/rclone/tree/fix-6547-mailru-times) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
`--dump bodies` uses up lots of memory.

Try the sync without `--dump headers,bodies` and hopefully it will work ok!

**Comment by dmak** [comment]
Yes, it worked smoothly. I confirm the issue of this ticket is solved with the suggested fix.

> I had an experiment and it seems mailru does support files before 1970-01-01

I am a bit concerned as I should not have files created before 1970-01-01 on local drive. Only one that is created on that date:
```
$ find . -mtime +10000 -exec ls -l '{}' \; | less
-rw-r----- 1 user users 1970770 Jan  1  1970   20220910-WA0005.jpeg
```

**Comment by ncw** [comment]
Thank you for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.60.1

> I am a bit concerned as I should not have files created _before_ 1970-01-01 on local drive. Only one that is created on that date:

Your file appears to be 1s before 1970-01-01 00:00:00 according to the log which I thought was odd. Maybe there is a rounding error or something!

Try `find . -mtime +10000 -exec ls -l --full-time'{}' \; | less` and see what that says.

**Comment by dmak** [comment]
Nevermind, I have already corrected the datetime for that file, so I cannot check anymore.
